File: src/main/java/com/manson/domain/fo76/VendingData.java
package com.manson.domain.fo76;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.manson.domain.AbstractObject;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = false)
@ToString(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class VendingData extends AbstractObject {

    private Boolean isVendedOnOtherMachine;
    private Integer price;
    private Integer machineType;

}

File: src/main/kotlin/com/manson/domain/fo76/items/VendingData.kt
package com.manson.domain.fo76.items

import com.manson.AbstractObject

open class VendingData : AbstractObject() {
    var vendedOnOtherMachine: Boolean? = null
    var price: Int? = null
    var machineType: Int? = null



    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is VendingData) return false
        if (!super.equals(other)) return false

        if (vendedOnOtherMachine != other.vendedOnOtherMachine) return false
        if (price != other.price) return false
        if (machineType != other.machineType) return false

        return true
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + (vendedOnOtherMachine?.hashCode() ?: 0)
        result = 31 * result + (price ?: 0)
        result = 31 * result + (machineType ?: 0)
        return result
    }

    override fun toString(): String {
        return "VendingData(vendedOnOtherMachine=$vendedOnOtherMachine, price=$price, machineType=$machineType)"
    }


}